### PR TITLE
Integration with obstacle design module

### PR DESCRIPTION
**Pull request by yang-zj1026**

### CODE REVIEW

File: gym_art/quadrotor_multi/quad_utils.py [hunk 1]
```diff
@@ -351,33 +362,33 @@ def perform_collision_between_drones(dyn1, dyn2):
...
+    #     new_shift_pos[1] += np.random.uniform(low=shift_pos[1] * -0.4, high=shift_pos[1] * 0.4)
+    #     new_shift_pos[2] = np.random.uniform(low=-1.3, high=-0.7)
+    #     if np.dot(shift_pos[:2], new_shift_pos[:2]) > 0:
+    #         shift_pos = new_shift_pos
+    #         break
+    #
+    # direction_mag = np.linalg.norm(shift_pos)
+    # direction_norm = shift_pos / (direction_mag + 0.00001 if direction_mag == 0.0 else direction_mag)
+    #
+    # drone_dyn.vel = direction_norm * drone_vel_mag
```
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/quad_utils.py:391]: Why we don't change vel here?

File: gym_art/quadrotor_multi/quad_utils.py [hunk 2]
```diff
@@ -323,13 +322,25 @@ def compute_col_norm_and_new_velocities(dyn1, dyn2):
...
+def compute_col_norm_and_new_vel_obst(dyn, obstacle_pos):
+    collision_norm = dyn.pos - obstacle_pos
+    coll_norm_mag = np.linalg.norm(collision_norm)
+    collision_norm = collision_norm / (coll_norm_mag + 0.00001 if coll_norm_mag == 0.0 else coll_norm_mag)
+
+    # Get the components of the velocity vectors which are parallel to the collision.
+    # The perpendicular component remains the same.
+    vnew = np.dot(dyn.vel, collision_norm)
+
+    return vnew, collision_norm
```
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/quad_utils.py:334]: Is this function used in anywhere?

**Review by Zhehui-Huang — CHANGES_REQUESTED** [review]
Overall, it is good. Please check my comments and fixed them.

**Review by Zhehui-Huang — CHANGES_REQUESTED** [review]
Overall, it is good. Please check my comments and fixed them.

**Comment by yang-zj1026** [comment]
Fixed.

**Review by Zhehui-Huang — CHANGES_REQUESTED** [review]
For both collisions between drones and collisions between drones and obstacles. I would recommend you check add coeff for noise as well.
For example
```
    cons_rand_val = np.random.normal(0, 0.8, 3)
    dyn1.vel += col_coeff * (cons_rand_val + np.random.normal(0, 0.15, 3))
    dyn2.vel += col_coeff * (-cons_rand_val + np.random.normal(0, 0.15, 3))
```

**Comment by yang-zj1026** [comment]
Fixed

**Review by Zhehui-Huang — APPROVED** [review]

**Pull request merged**